core/src/main/java/org/infinispan/topology/CacheTopology.java

```diff
@@ -123,6 +126,22 @@ else if (currentCH != null)
     * Read operations should always go to the "current" owners.
     */
    public ConsistentHash getReadConsistentHash() {
+      switch (phase) {
+         case STABLE:
+            assert pendingCH == null;
+            assert unionCH == null;
+            return currentCH;
+         case READ_OLD_WRITE_ALL:
+            assert pendingCH != null;
+            assert unionCH != null;
+            return currentCH;
+         case READ_ALL_WRITE_ALL:
+            assert pendingCH != null;
+            return unionCH;
+         case READ_NEW_WRITE_ALL:
+            assert unionCH != null;
+            return pendingCH;
```

Comment:
I think this would be a good time to rename `currentCH` to `oldCH` and `pendingCH` to `newCH`. But maybe in a separate PR :)